pageserver/src/tenant/timeline.rs

```diff
@@ -897,15 +896,12 @@ impl Timeline {
         Ok(())
     }
 
-    pub fn activate(self: &Arc<Self>, ctx: &RequestContext) -> anyhow::Result<()> {
-        if is_broker_client_initialized() {
-            self.launch_wal_receiver(ctx, get_broker_client().clone())?;
-        } else if cfg!(test) {
-            info!("not launching WAL receiver because broker client hasn't been initialized");
-        } else {
-            anyhow::bail!("broker client not initialized");
-        }
-
+    pub fn activate(
+        self: &Arc<Self>,
+        broker_client: &'static BrokerClientChannel,
+        ctx: &RequestContext,
+    ) -> anyhow::Result<()> {
+        self.launch_wal_receiver(ctx, (*broker_client).clone())?;
```

Comment:
Probably similarly this clone could be removed? Nevermind, an owned BrokerClientChannel is needed. Maybe this:

```suggestion
        self.launch_wal_receiver(ctx, broker_client.to_owned())?;
```

Reply:
Hmm this is further changed in https://github.com/neondatabase/neon/pull/4319/files#r1203774737

Reply:
I think this is fine. Perhaps there was no point in differentiating between `&'static _` and `_` after all.